abstractions/php/src/HttpCore.php

```diff
@@ -0,0 +1,9 @@
+<?php
+namespace Microsoft\Kiota\Abstractions;
+
+use Http\Promise\Promise;
+
+interface HttpCore {
+    public function sendAsync(RequestInfo $requestInfo, object $targetClass, ResponseHandler $responseHandler): Promise;
+    public function sendPrimitiveAsync(RequestInfo $requestInfo, object $targetClass, ResponseHandler $responseHandler): Promise;
```

Comment:
the send Primitive Async method is only needed because there is a "big" difference, between value types and object types in java/dotnet. Might not be needed for PhP. Especially since the Promise type is not generic to the target type.